app/Jobs/UserRatePackage.php

```diff
@@ -50,4 +57,14 @@ public function handle()
             $rating->save();
         }
     }
+
+    private function isSelfAuthored($package)
+    {
+        return (int) $package->author_id === (int) $this->userId;
+    }
+
+    private function isSelfContributed($package)
+    {
+        return $package->contributors->pluck('user_id')->contains($this->userId);
```

Comment:
Could we up the performance a bit by doing `$package->contributors()->pluck('user_id')`? I vaguely think that `pluck` on an Eloquent query is equivalent to a (SELECT that thing then pluck that thing on the result) but I can't say 100% for sure.

Reply:
Looks good to me!  I just changed it.